packages/fast-components-react-base/src/dialog/dialog.tsx

```diff
@@ -163,16 +167,12 @@ class Dialog extends Foundation<DialogHandledProps, DialogUnhandledProps, {}> {
     };
 
     /**
-     * handles mouse clicks on modal overlay
+     * Handle overlay touch and clicks
      */
-    private handleOverlayClick = (event: React.MouseEvent): void => {
-        if (
-            this.props.onDismiss &&
-            typeof this.props.onDismiss === "function" &&
-            this.props.visible
-        ) {
-            this.props.onDismiss(event);
-        }
+    private handleOverlaySelection = (
+        event: React.TouchEvent | React.MouseEvent
```

Comment:
why do you need this method? Seems since you're simply invoking `checkForSoftDismiss` with the event object that you can just provide `this.checkForSoftDismiss` as the listener

Reply:
yep